### PR TITLE
[Chore](build) fix some undefined behavior about incomplete type vector

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes
incomplete type vector is ub, this pr will fix it.
https://www.lukas-barth.net/blog/cpp20-vector-incomplete/
https://reviews.llvm.org/D68365
图片

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: thirdparty/patches/rocksdb-5.14.2.patch [hunk 1]
```diff
@@ -9,3 +9,88 @@
...
    bool eof_;
  };
+diff --git a/include/rocksdb/metadata.h b/include/rocksdb/metadata.h
+--- a/include/rocksdb/metadata.h
++++ b/include/rocksdb/metadata.h
+@@ -15,42 +15,6 @@
+ 
+ namespace rocksdb {
+ struct ColumnFamilyMetaData;
+-struct LevelMetaData;
```
Comment (yiguolei) [inline thirdparty/patches/rocksdb-5.14.2.patch:19]: I did not find any changes. I find you only change the sequence of these code, including thrift defenitions. Anything wrong in previous code?

Comment (BiteTheDDDDt) [reply thirdparty/patches/rocksdb-5.14.2.patch:19]: By changing the declaration order, the forward declaration is removed, then T is not incomplete type when defining vector<T>.

Comment (BiteTheDDDDt) [reply thirdparty/patches/rocksdb-5.14.2.patch:19]:
```cpp
struct Inner;
struct S {
  std::vector<Inner> v;
};

struct Inner {
  int a;
};
```
This is a UB example.

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**